homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,579 @@
+    LEFT JOIN job_history JH 
+        ON E.employee_id = JH.employee_id
+    WHERE JH.employee_id IS NUlL
+
+
 --30. the employee number, name( first name and last name ) and job title for all employees whose salary is more than any average salary of any department.
+
+SELECT employee_id, 
+    last_name || ' ' || first_name AS "FULL NAME",
+    job_title
+FROM employees
+    JOIN jobs USING (job_id)
+    WHERE salary > (
+        SELECT MAX(AVG(salary))
+        FROM employees
+            JOIN departments USING (department_id)
+            GROUP BY department_id
+    );
+	
 --31. the employee id, name ( first name and last name ) and the job id column with a modified title SALESMAN for those employees whose job title is ST_MAN and DEVELOPER for whose job title is IT_PROG.
+
+SELECT employee_id,
+    first_name || ' ' || last_name AS "FULL NAME",
+    case job_id
+        WHEN 'ST_MAN' THEN 'SALESMAN'
+        WHEN 'IT_PROG' THEN 'DEVELOPER'
+    ELSE job_id
+    END AS "JOB"
+FROM employees;
+
 --32. the employee id, name ( first name and last name ), salary and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than the average salary of all employees.
+
+SELECT employee_id,
+    first_name || ' ' || last_name AS "FULL NAME",
+    salary,
+    CASE
+        WHEN salary < (
+            SELECT avg(salary) 
+            FROM employees
+        ) THEN 'LOW'
+        WHEN salary > (
+            SELECT avg(salary) 
+            FROM employees
+        ) THEN 'HIGH'
+    ELSE NULL
+    END AS "SALARY STATUS"
+FROM employees;
+    
+
 --33. the employee id, name ( first name and last name ), SalaryDrawn, AvgCompare (salary - the average salary of all employees)
     -- and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than
     -- the average salary of all employees.
+	
+SELECT employee_id,
+    first_name || ' ' || last_name AS "FULL NAME",
+    salary,
+    (
+        SELECT ROUND(AVG(salary))
+        FROM employees
+    ) AS "AVG COMPARE",
+    CASE
+        WHEN salary < (
+            SELECT AVG(salary) 
+            FROM employees
+        ) THEN 'LOW'
+        WHEN salary > (
+            SELECT AVG(salary) 
+            FROM employees
+        ) THEN 'HIGH'
+    ELSE NULL
+    END AS "SALARY STATUS"
+FROM employees;
+
 --34. all the employees who earn more than the average and who work in any of the IT departments.
+
+SELECT *
+FROM employees
+    JOIN departments USING (department_id)
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+    )
+    AND department_name LIKE '%IT%'
+	
 --35. who earns more than Mr. Ozer.
+
+SELECT *
+FROM employees
+    WHERE salary > (
+        SELECT salary
+        FROM employees
+            WHERE last_name || ' ' || first_name LIKE '%Ozer%';
+    )
+
 --36. which employees have a manager who works for a department based in the US.
+
+SELECT E.*
```

Comment:
incorect query, 122 manager doesn't work in US region, but I can see the employee with this manager